### ISSUE TITLE
ensureFile do not create file if there is a folder of the same name at given path

### ISSUE THREAD

**Issue by devrafalko** [jprichardson/node-fs-extra#470]
```
fs.ensureFile('./dist/new.js',(err)=>{
  console.log('file: ',err);
});
```

If there is `./dist/new.js` **folder** at this path, the `ensureFile` method do not create `new.js` file and do not throw any error in callback function: `err:undefined`.
Vice versa, if there is `./dist/new.js` **file** and I attempt to add folder of the same name and path with `ensureDir`, this method correctly throws error in the callback function: `Error: EEXIST: file already exists.`

**Comment by RyanZim** [comment]
This was fixed by https://github.com/jprichardson/node-fs-extra/pull/466, but it's not released yet. I'll try to release a new version ASAP.

**Comment by RyanZim** [comment]
Released in fs-extra 4.0.1 :tada:

**Comment by devrafalko** [comment]
great! thank you 👍
